### PR TITLE
Darwin builds - Closes #2902

### LINKED ISSUE #2902: Builds for Darwin
### Expected behavior

It is possible to create Darwin builds using MacOS

### Actual behavior

Only Linux is supported at the moment.

### Steps to reproduce

Follow instructions in `build/README.md`

### Which version(s) does this affect? (Environment, OS, etc...)

Binary builds up to 1.4

### PR DESCRIPTION
**Pull request by ghost**
### What was the problem?

It was not possible to create builds for macOS.

### How did I fix it?

Move the linux-specific logic from `build.sh` to `Makefile` splitting OS-specifics in included `*.mk`  files.

### How to test it?

Using macOS `10.12` or `10.13`, follow the instructions in `build/README.md`; e.g.:
```
# brew install coreutils gpg2 jq moreutils
cd build/
make LISK_NETWORK=testnet
```

### Review checklist

* The PR resolves #2902 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

**Review by psychopenguin — APPROVED** [review]

File: build/README.md [hunk 1]
```diff
@@ -12,6 +12,20 @@ The following packages must be installed to build releases with lisk-build:
 sudo apt-get install autoconf build-essential git jq libtool moreutils python2.7 tcl8.5 wget zlib1g-dev
 ```
 
+## MacOS
+
+A nachine running macOS 10.12 or 10.13 is assumed.
```
Comment (ManuGowda) [inline build/README.md]: typo `machine`

Comment (fchavant) [reply build/README.md]: Good catch.

**Review by ManuGowda — APPROVED** [review]

**Review by Nazgolze — APPROVED** [review]

File: build/README.md [hunk 2]
```diff
@@ -12,6 +12,20 @@ The following packages must be installed to build releases with lisk-build:
 sudo apt-get install autoconf build-essential git jq libtool moreutils python2.7 tcl8.5 wget zlib1g-dev
 ```
 
+## MacOS
+
+A machine running macOS 10.12 or 10.13 is assumed.
```
Comment (MaciejBaj) [inline build/README.md]: is required?

Comment (fchavant) [reply build/README.md]: MacOS is required. We guarantee it works on 10.12 or 10.13; earlier version might or might not work.

Comment (fchavant) [reply build/README.md]: @MaciejBaj should this be reworded?

**Review by Nazgolze — APPROVED** [review]

**Review by MaciejBaj — APPROVED** [review]

**Pull request merged**